demux/utils/novaseq_samplesheet.py

```diff
@@ -0,0 +1,203 @@
+        return delimiter in index
+
+    @staticmethod
+    def is_dummy_sample_in_samplesheet(dummy_index: str, sample_indexes: list) -> bool:
+        """ Determines if a dummy sample is already present in the samplesheet """
+        return any(
+            sample_index.startswith(dummy_index) for sample_index in sample_indexes
+        )
+
+    @staticmethod
+    def sample_indexes_in_lane(samplesheet: list, lane: str) -> list:
+        """ Returns all sample indexes in a given lane """
+        return [sample["index"] for sample in samplesheet if sample["lane"] == lane]
+
+    def is_reverse_complement(self) -> bool:
+        """If the run used the new NovaSeq control software version (1.7.0) and the new
+        reagent kit version (1.5) the second index should be the reverse complement"""
+        return (
+            self.runparameters.control_software_version
+            == self.NEW_CONTROL_SOFTWARE_VERSION
+            and self.reagent_kit_version() == self.NEW_REAGENT_KIT_VERSION
+        )
+
+    def add_dummy_indexes(self, raw_samplesheet) -> "CreateNovaseqSamplesheet":
+        """Add all dummy indexes to raw sample sheet. Dummy indexes are used to check for index
+        contamination"""
+        with open(f"{self.dummy_indexes_file}") as csv_file:
+            dummy_samples_csv = csv.reader(csv_file, delimiter=COMMA)
+            dummy_samples = [row for row in dummy_samples_csv]
+            added_dummy_samples = []
+            lanes = {sample["lane"] for sample in raw_samplesheet}
+
+            for lane in lanes:
+                sample_indexes = self.sample_indexes_in_lane(raw_samplesheet, lane)
+                for name, dummy_index in dummy_samples:
+                    if not self.is_dummy_sample_in_samplesheet(
+                        dummy_index, sample_indexes
+                    ):
+                        add_dummy_sample = self.add_dummy_sample(
+                            self.flowcell, dummy_index, lane, name
+                        )
+                        added_dummy_samples.append(add_dummy_sample)
+
+            self.raw_samplesheet.extend(added_dummy_samples)
+
+            return raw_samplesheet
+
+    def remove_unwanted_indexes(self, raw_samplesheet) -> "CreateNovaseqSamplesheet":
+        """ Filter out indexes of unwanted length and single indexes """
+
+        raw_samplesheet = [
+            line for line in raw_samplesheet if self.is_dual_index(line["index"])
+        ]
+        return raw_samplesheet
+
+    def adapt_indexes(self, raw_samplesheet) -> "CreateNovaseqSamplesheet":
+        """Adapts the indexes: pads all indexes so that all indexes have a length equal to the
+        number  of index reads, and takes the reverse complement of index 2 in case of the new
+        novaseq software control version (1.7) in combination with the new reagent kit
+        (version 1.5)"""
+
+        reverse_complement = self.is_reverse_complement()
```

Comment:
```suggestion
        is_reverse_complement = self.is_reverse_complement()
```
Otherwise it awfully similar to the def call `def reverse_complement(dna: str) -> str:`
Also prepending with `is_` makes one think boolean (at least me)

Reply:
I was worried about name collisions but I was wrong. Fixed.